### ISSUE TITLE
Fetch not bound to window

### ISSUE THREAD

**Issue by joshuahaertel** [badgateway/oauth2-client#105]
I am getting the following error in a browser: `TypeError: Failed to execute 'fetch' on 'Window': Illegal invocation at t.OAuth2Client.request`

The problem appears to be that fetch binds to the settings object but it needs to be bound to the window. Probably the easiest fix is to pass the fetch call as a lambda in the client constructor like this:

```
const originalFetch = fetch;
const client = new OAuth2Client({
  ...
  fetch: (...args) => originalFetch(...args),
});
```

I do not think this is a problem when running in a node environment. I am not sure how this plays out in a service worker.

The best fix is probably to change this:

https://github.com/badgateway/oauth2-client/blob/fdd7f7547d64c280c8f0aab53a54028f8269f650/src/client.ts#L104 

to something like what I've done in the constructor.

As a side note, I use a cached copy of the original fetch because I think it's best practice for all oauth related calls to protect against XSS. [This](https://twitter.com/coffeetocode/status/1312998927881314305) provides a good explanation of why, though the next comment shows that there's still more a hacker could do. That's why it's even better to make sure this code can run in a properly implemented service worker.

**Comment by fstuff** [comment]
Thank you, I experienced the same issue and was able to resolve it with your suggestion to pass the original fetch into the client settings as a lambda.  

> I do not think this is a problem when running in a node environment. I am not sure how this plays out in a service worker.
FWIW my environment is using node 18 with webpack, but the issue may be caused by something elsewhere in my config or environment--I'm new to using this library.

**Comment by jozefizso** [comment]
This happens to me with Edge 114.

**Comment by glemmaPaul** [comment]
Same here! Downgrading to 2.0.18 worked for me.

**Comment by evert** [comment]
Thanks for the all the reports. I wish it was a bit easier to have tests run in a browser context with CI.

**Comment by evert** [comment]
Released as 2.2.1
